open-apparel-registry/open-apparel-registry
# Add a method to build an index table for a newly trained model

## Overview
To ensure the index table matches the trained model, we're now creating a new index table unique to the trained model and populating it immediately after training.

Connects #2035

## Testing Instructions

* run `./scripts/manage migrate`
* run `./scripts/manage train_model`
* run `./scripts/dbshell`
* in the database shell, run this query to get the index of the most recently trained model:
```
select id, created_at, is_active, activated_at from api_trainedmodel;
 id |          created_at           | is_active | activated_at
----+-------------------------------+-----------+--------------
  1 | 2022-08-25 19:34:42.750128+00 | f         |
  2 | 2022-08-25 19:38:29.710412+00 | f         |
  3 | 2022-08-25 19:39:19.838907+00 | f         |
  4 | 2022-08-25 19:46:18.925169+00 | f         |
  5 | 2022-08-25 19:49:51.29007+00  | f         |
  6 | 2022-08-25 19:58:20.242987+00 | f         |
  7 | 2022-08-25 20:04:50.594806+00 | f         |
  8 | 2022-08-25 20:07:26.114378+00 | f         |
  9 | 2022-08-25 20:13:37.480748+00 | f         |
(9 rows)
```
* to verify that the index table has been created, run the following command in the postgres shell, replacing 9 with the id of the most recently trained model:
```
\d dedupe_indexed_records_9
               Table "public.dedupe_indexed_records_9"
   Column    |         Type          | Collation | Nullable | Default
-------------+-----------------------+-----------+----------+---------
 block_key   | text                  |           |          |
 record_id   | character varying(32) |           |          |
 record_data | text                  |           |          |
Indexes:
    "dedupe_indexed_records_9_block_key_record_id_key" UNIQUE CONSTRAINT, btree (block_key, record_id)
```

* run the query below to verify that there are rows in the index table
```
openapparelregistry=# select count(*) from dedupe_indexed_records_9;
 count
-------
   146
(1 row)
```

## Checklist
